src/webextension/survey/components/fields/reason.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { formValueSelector } from 'redux-form';
+
+export const REASONS = [
+  { min: 4, value: 'fast', name: 'It is fast' },
+  { max: 4, value: 'slow', name: 'It is slow' },
+  { min: 4, value: 'works', name: 'It works well' },
+  { max: 4, value: 'broken', name: 'It is broken' },
+  { min: 4, value: 'like', name: 'I like it' },
+  { max: 4, value: 'dislike', name: 'I do not like it' },
+  { value: 'other', name: 'Other' }
+];
+
+class Reason extends Component {
+  shouldRender() {
+    const { max, min, sentiment } = this.props;
+    return min && sentiment >= min ||
+      max && sentiment <= max ||
+      (!sentiment || !max && !min);
+  }
+
+  render() {
+    if (this.shouldRender()) {
+      const { name, value } = this.props;
+      return <option value={value}>{name}</option>;
+    }
+    return null;
+  }
+}
+
+Reason.propTypes = {
+  max: React.PropTypes.number,
+  min: React.PropTypes.number,
+  name: React.PropTypes.string.isRequired,
+  sentiment: React.PropTypes.number,
+  value: React.PropTypes.string.isRequired
+};
+
+class ReasonField extends Component {
+  renderSelect() {
+    const { input, sentiment } = this.props;
+    return (
+      <select {...input}>
+        <option>Select one...</option>
+        {
+          REASONS.map(reason => (
+            <Reason sentiment={sentiment} key={reason.value} {...reason} />
+          ))
+        }
+      </select>
+    );
+  }
+
+  render() {
+    const { input, sentiment } = this.props;
+    if (!sentiment) {
+      return null;
+    }
+    return (
+      <div className="field{ this.required }">
```

Comment:
Looks like this `className` may not be interpolating:

![pulse_and_builds_user_sentiment_survey__closes__10__closes__18___by_chuckharmston_ _pull_request__82_ _mozilla_pulse_and_1____npm_bin__jpm_run_--addon-dir_build_-b_nightly__firefox-bin__and_index_js_ _build_webextension_survey_ _pulse](https://cloud.githubusercontent.com/assets/557895/22609499/45a5f310-ea16-11e6-83ab-01b9f21d66fc.png)